.github/workflows/pr_build.yml

```diff
@@ -46,7 +46,7 @@ jobs:
         os: [ubuntu-latest]
         java_version: [8, 11, 17]
         test-target: [rust, java]
-        spark-version: ['3.4']
+        spark-version: ['3.5']
```

Comment:
I think we still need to keep 3.4 here as it is the default Spark version. Otherwise, PRs won't be tested with Spark3.4.

Or was this intentional because Mac tests are still using Spark 3.4?

Reply:
We have `linux-test` and `linux-test-with-old-spark`, which I thought both run against PRs, but maybe I am misunderstanding this.

I updated `linux-test` to use 3.5 instead of 3.4 and added 3.4 to `linux-test-with-old-spark`.

Reply:
My mistake, you are right. They both run against PRs.

So should we update the Mac ones to similarly, then?